### PR TITLE
feat(optimizer): change create mv field name

### PR DESCRIPTION
**Pull request by st1page**
## What's changed and what's your intention?
- remove the row id number
- if a create materialized view generate multiple same name of a hidden column, it will add a number.
 
#1363 not really solve the problem, except row id, all columns generated in pk deriving could be not unique.
you can see `src/frontend/test_runner/tests/testdata/pk_derive.yaml` for detailed case

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

### CODE REVIEW

**Comment by xiangjinwu** [comment]
Seems related to #2467, where a unique field name is preferred, and the duplicates originated from different PlanNode.
* In #2467, the 2 `expr#0`s are from 2 different LogicalProjects, one before LogicalAgg and one after LogicalAgg. I tried to dedup them by prepending PlanNodeId but failed to find an elegant way.
* The difference is, #2467 only affects field names for debugging. This PR is concerned about column names in catalog, although they are hidden and not selectable by user.
* As for tumble window, they already have a user-visible name in bind context, which will be propagated to PlanRoot and then mview column name. But it is not available as schema field name on intermediate PlanNodes.

**Comment by st1page** [comment]
so we have two kinds of field names
- user-visible field names, which are declared in the select clause.
- internal-schema field names, which are for debugging and just used in our system 

but in create materialized view statement we use both of them. and we should ensure all column name in materialized view is unique
- for user-visible column part we can force user to use alias and make sure they are unique.
- for the hidden columns we just use the internal-schema field names, but in materialized operator gives the materialized view columns a unique name by give number. And we will not guarantee these columns name. for example a user give `create materialized mv1 view select max(v) from t group id;` we might give an error that mv1 does not have column id.

File: src/frontend/src/optimizer/plan_node/stream_materialize.rs [hunk 1]
```diff
@@ -132,10 +110,18 @@ impl StreamMaterialize {
...
+                } else {
+                    match col_names.try_insert(field.name.clone(), 0) {
+                        Ok(_) => field.name.clone(),
+                        Err(mut err) => {
+                            let cnt = err.entry.get_mut();
+                            *cnt += 1;
+                            field.name.clone() + "#" + &cnt.to_string()
+                        }
+                    }
+                };
```
Comment (st1page) [inline src/frontend/src/optimizer/plan_node/stream_materialize.rs:124]: the core change is here.

**Review by xiangjinwu — APPROVED** [review]

File: src/frontend/src/catalog/mod.rs [hunk 2]
```diff
@@ -47,8 +47,8 @@ pub fn check_valid_column_name(column_name: &str) -> Result<()> {
 
 const ROWID_PREFIX: &str = "_row_id";
 
-pub fn gen_row_id_column_name(idx: usize) -> String {
-    ROWID_PREFIX.to_string() + "#" + &idx.to_string()
+pub fn row_id_column_name() -> String {
+    ROWID_PREFIX.to_string()
```
Comment (st1page) [inline src/frontend/src/catalog/mod.rs:51]: 😆

**Pull request merged**